signac/_utility.py

```diff
@@ -122,7 +122,8 @@ def _dotted_dict_to_nested_dicts(dotted_dict, delimiter_nested="."):
 
 class _hashable_dict(dict):
     def __hash__(self):
-        return hash(tuple(sorted(self.items())))
+        hashed = {_to_hashable(k): _to_hashable(v) for k, v in self.items()}
```

Comment:
I'm not totally sure if we need to hash the key, because we already put restrictions on state point keys

Reply:
Python’s dict implementation requires that dict keys must be hashable. We can skip making the keys hashable.

Perhaps this can be a one-liner (may require reformatting):
```python
return hash(tuple(sorted((k, _to_hashable(v)) for k, v in self.items())))
```